src/app/modules/spcp/spcp.oidc.client.ts

```diff
@@ -169,8 +183,10 @@ export class SpOidcClient {
       scope: 'openid',
       response_type: 'code',
       state: state,
-      esrvc: esrvcId,
       nonce: ulid(), // Not used - nonce is a required parameter for SPCP's OIDC implementation although it is optional in OIDC specs
+      ...(this.#authType === FormAuthType.SP
+        ? { esrvc: esrvcId }
+        : { esrvcID: esrvcId }), // Additional parameter for agencies' eservice Id. Labelled esrvc in Singpass OIDC and esrvcID in Corppass OIDC
```

Comment:
This (being given inconsistent naming specs by NDI for corpPass and singPass) is so damn sad 😑 

/suggestionFeelFreeToIgnore (also not sure if Typescript will allow):
```suggestion

      // Additional parameter for agencies' eservice Id. Labelled esrvc in Singpass OIDC and esrvcID in Corppass OIDC
      [this.#authType === FormAuthType.SP ? 'esrvc' : 'esrvcID']: esrvcId,
```

Reply:
Fixed in #4516